### PR TITLE
mrd-2293_bulk-delete-notes-fix: added param to findByCrn to allow ret…

### PR DESCRIPTION
**Pull request by bills7281**
…rieval of deleted records, also altered schedule to ensure bulk delete runs OOH

### CODE REVIEW

**Review by robinandrewdevilliers — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt [hunk 1]
```diff
@@ -43,12 +43,12 @@ internal class DeleteRecommendationService(
   suspend fun getSystemDeleteRecommendationResponse(
     crn: String,
   ): ResponseEntity<DeleteRecommendationResponse> {
-    val recommendations = recommendationRepository.findByCrn(crn).sorted()
+    val recommendations = recommendationRepository.findByCrn(includeDeleted = true, crn = crn).sorted()
     return ResponseEntity(
       DeleteRecommendationResponse(
-        sensitive = recommendations.firstOrNull()?.data?.sensitive ?: false,
+        sensitive = recommendations[0].data.sensitive ?: false,
```
Comment (DarrenW-HMPPS) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt:49]: Isn't this going to throw an exception if recommendations is empty?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationsCleanupTask.kt [hunk 2]
```diff
@@ -18,7 +18,8 @@ internal class RecommendationsCleanupTask(
   @Value("\${mrd.api.url}") private val mrdApiUrl: String? = null,
   private val environment: Environment? = null,
 ) {
-  @Scheduled(initialDelay = 0, fixedDelay = 1814400000) // Initial delay: 0 milliseconds, Fixed delay: 21 days
+  @Scheduled(initialDelay = 0, fixedRate = 1814400000) // Initial delay: 0 milliseconds, Fixed rate: 21 days
+  @Scheduled(cron = "0 30 19 1/21 * ?") // The second schedule will run at 7:30 PM every 21 days starting from the first day of the month
```
Comment (DarrenW-HMPPS) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationsCleanupTask.kt:22]: Are you sure about this? I don't think that's valid.
First value should be minutes I think. 
Unless this is a variation on the standard cron format.

**Review by DarrenW-HMPPS — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationsCleanupTask.kt (hunk 2 above)
Comment (DarrenW-HMPPS) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationsCleanupTask.kt:22]: Ignore me, it's OK.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt (hunk 1 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt:49]: The crn comes from a prior database call in RecommendationsCleanupTask so at this point we already know it exists.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt [hunk 3]
```diff
@@ -43,12 +43,12 @@ internal class DeleteRecommendationService(
...
         notes = "Recommendation expired, deleted by system\n" +
-          "View the case summary for ${recommendations.firstOrNull()?.data?.personOnProbation?.name}: $mrdUrl/cases/${recommendations.firstOrNull()?.data?.crn}/overview",
+          "View the case summary for ${recommendations[0].data.personOnProbation?.name}: $mrdUrl/cases/${recommendations[0].data.crn}/overview",
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt:51]: The crn comes from a prior database call in RecommendationsCleanupTask so at this point we already know it exists.

**Review by DarrenW-HMPPS — APPROVED** [review]

**Pull request merged**